### PR TITLE
`tbl_svysummary(include=)` updates to include more of the design columns

### LINKED ISSUE #1166: Bug Report: stratified tables with tbl_svysummary() not always working
hi folks - I might be getting things entirely wrong here, but tbl_svysummary() throws an error if I try to create a grouped table (using `by = `) when the grouping variable of interest is the `strata` variable in the survey design. 
Context: for a stratified survey (say e.g. want to get the percentage of people vaccinated in two different regions of the same district - or say two subdivisions of a refugee camp) - we would take an appropriate sample from each strata, then the survey design would include the regions as the stratifying variable, and then we would be able to calculate stratified estimates for each region using the `by` argument (as well as being able to calculate an appropriately weighted overall estimate when not using `by`). 
An example below comparing to {survey} and {srvyr} outputs below using the api dataset. (Example 1)
As well as an example of the original error that was thrown. (Example 2)
(Sorry is not a full reprex - markdown formatting messes up the output)

Also sorry for lengthy post but just trying to explain as much as possible - thanks a lot for all the work you do on this package! 

- In example 1, things work when I use {srvyr} to select variables in advance of `tbl_svysummary()` and outputs are the same as when using only {survey} functions;  but the error I get from `survey_design %>% tbl_svysummary(by = "stype", include = c(both, stype))` is: 
```r
Error in `by=` argument input. Select from ‘cds’, ‘name’, ‘sname’, ‘snum’, ‘dname’, ‘dnum’, ‘cname’, ‘cnum’, ‘flag’, ‘pcttest’, ‘api00’, ‘api99’, ‘target’, ‘growth’, ‘sch.wide’, ‘comp.imp’, ‘both’, ‘awards’, ‘meals’, ‘ell’, ‘yr.rnd’, ‘mobility’, ‘acs.k3’, ‘acs.46’, ‘acs.core’, ‘pct.resp’, ‘not.hsg’, ‘hsg’, ‘some.col’, ‘col.grad’, ‘grad.sch’, ‘avg.ed’, ‘full’, ‘emer’, ‘enroll’, ‘api.stu’
```
- In example 2 when running `survey_design %>%  select(died, health_district) %>%  tbl_svysummary(by = died)` I get the following error, but the other methods seem to work fine: 
```r
Error: Problem with `mutate()` column `stat_label`.
i `stat_label = stat_label_match(.data$stat_display)`.
x Column `stat` not found in `.data`
Run `rlang::last_error()` to see where the error occurred.
> rlang::last_error()
x
+-<error/dplyr:::mutate_error>
| Problem with `mutate()` column `stat_label`.
| i `stat_label = stat_label_match(.data$stat_display)`.
| x Column `stat` not found in `.data`
\-<error/rlang_error_data_pronoun_not_found>
  Column `stat` not found in `.data`
Backtrace:
  1. survey_design %>% select(died, health_district) %>% tbl_svysummary(by = died)
 27. rlang:::abort_data_pronoun(x)
Run `rlang::last_trace()` to see the full context.
```

-----------------------------------------------------------------------

```r
## install extra packages from github (for example 2)
# remotes::install_github("R4EPI/epitabulate")
# remotes::install_github("R4EPI/epidict")
# remotes::install_github("R4EPI/epikit")

## load packages
library("dplyr")       # for wrangling
library("epitabulate") # for dplyr wrappers of survey package
library("epidict")     # for fake datasets
library("epikit")      # for calculating weights
library("survey")      # for survey functions
library("srvyr")       # dplyr wrapper for survey package
library("gtsummary")   # produce tables)



## EXAMPLE 1

    
    
# A dataset with a complex design
data(api, package = "survey")

## define a survey object from stratified
survey_design <- survey::svydesign(id=~1,strata=~stype, weights=~pw, data=apistrat, fpc=~fpc)

## make {srvyr} object for reference
survey_design_srvyr <- srvyr::as_survey(survey_design)


## calculate proportions using {survey} object and {gtsummary}
survey_design %>%
  tbl_svysummary(by = "stype", include = c(both, stype))

## calculate proportions using {srvyr} object and {gtsummary}
survey_design_srvyr %>%
  select(both, stype) %>%
  tbl_svysummary(by = "stype")

## calculate proportions using {srvyr} object and {epitabulate}
survey_design_srvyr %>%
  tab_survey(both, strata = stype)

## calculate proportions using {survey} object and {survey} funs
svyby(~both, by = ~ stype, FUN = svyciprop, design = survey_design, vartype = "ci")





##  EXAMPLE 2


## creating fake data, population and cluster counts

## generates a fake dataset for use as an example
study_data_cleaned <- epidict::gen_data(dictionary = "Mortality",
                           varnames   = "name",
                           numcases   = 1000)


## create an age group variable by specifying categorical breaks (of years)
study_data_cleaned <- study_data_cleaned %>%
  mutate(age_group = age_categories(age_years,
                                    breakers = c(0, 3, 15, 30, 45)
  ))


## create a character variable based off groups of a different variable
study_data_cleaned <- study_data_cleaned %>%
  mutate(health_district = case_when(
    cluster_number %in% c(1:5) ~ "district_a",
    TRUE ~ "district_b"
  ))

## generate population data by age groups in years for district A
population_data_age_district_a <- gen_population(total_pop = 10000, # set total population
                                                 groups      = c("0-2", "3-14", "15-29", "30-44", "45+"), # set groups
                                                 proportions = c(0.0340, 0.1811, 0.1380, 0.0808, 0.0661), # set proportions for each group
                                                 strata      = c("male", "female")) %>%           # stratify by gender
  rename(age_group  = groups,                      # rename columns (NEW NAME = OLD NAME)
         sex        = strata,
         population = n) %>%
  mutate(health_district = "district_a")           # add a column to identify region


## generate population data by age groups in years for district B
population_data_age_district_b <- gen_population(total_pop = 10000, # set total population
                                                 groups      = c("0-2", "3-14", "15-29", "30-44", "45+"), # set groups
                                                 proportions = c(0.0340, 0.1811, 0.1380, 0.0808, 0.0661), # set proportions for each group
                                                 strata      = c("male", "female")) %>%           # stratify by gender
  rename(age_group  = groups,                      # rename columns (NEW NAME = OLD NAME)
         sex        = strata,
         population = n) %>%
  mutate(health_district = "district_b")           # add a column to identify region



## bind region population data together to get overall population
population_data_age <- bind_rows(population_data_age_district_a,
                                 population_data_age_district_b)


## create fake cluster countss
cluster_counts <- tibble(cluster = c("village_1", "village_2", "village_3", "village_4",
                                     "village_5", "village_6", "village_7", "village_8",
                                     "village_9", "village_10"),
                         households = c(700, 400, 600, 500, 300,
                                        800, 700, 400, 500, 500))

## adding survey weights

## get the number of people of individuals interviewed per household
## adds a variable with counts of the household (parent) index variable
study_data_cleaned <- study_data_cleaned %>%
  add_count(index, name = "interviewed")


## create a variable called "surv_weight_strata"
## contains weights for each individual - by age group, sex and health district
study_data_cleaned <- add_weights_strata(x = study_data_cleaned,
                                         p = population_data_age,
                                         surv_weight = "surv_weight_strata",
                                         surv_weight_ID = "surv_weight_ID_strata",
                                         age_group, sex, health_district)

## create cluster weights
study_data_cleaned <- add_weights_cluster(x = study_data_cleaned,
                                          cl = cluster_counts,
                                          eligible = member_number,
                                          interviewed = interviewed,
                                          cluster_x = village_name,
                                          cluster_cl = cluster,
                                          household_x = index,
                                          household_cl = households,
                                          surv_weight = "surv_weight_cluster",
                                          surv_weight_ID = "surv_weight_ID_cluster",
                                          ignore_cluster = FALSE,
                                          ignore_household = FALSE)

## create a survey weight for cluster and strata
study_data_cleaned <- study_data_cleaned %>%
  mutate(surv_weight_cluster_strata = surv_weight_strata * surv_weight_cluster)

## drop missing weights
study_data_cleaned <- study_data_cleaned %>%
  filter(!is.na(surv_weight_cluster_strata))


## create a srvyr object for a stratified cluster survey
survey_design <- study_data_cleaned %>%
  as_survey_design(ids = village_name, # cluster ids
                   weights = surv_weight_cluster_strata, # weight variable created above
                   strata = health_district # sampling was stratified by district
  )


## make died in to a TRUE/FALSE variable (because {survey} doesnt accept chars)
survey_design <- survey_design %>%
  mutate(died = died == "yes")


## calculate proportions using {survey} object and {survey} funs
svyby(~died, by = ~health_district, FUN = svyciprop, design = survey_design, vartype = "ci")


## calculate proportions using {srvyr} object and {epitabulate}
tab_survey(survey_design, died,
           strata = health_district,
           deff = TRUE)

## calculate proportions with {srvyr} object and {gtsummary}
survey_design %>%
  select(died, health_district) %>%
  tbl_svysummary(by = died)

```

### PR DESCRIPTION
**Pull request by ddsjoberg**
@larmarange are you able to review this?

**What changes are proposed in this pull request?**

* No longer removing the strata column from survey objects from the columns that will be summarized in `tbl_svysummary()`. If users previously didn't indicate which variables to summarize with `include=`, then the strata column (if present) will now appear in the summary table. When using a srvyr class object, all columns will be printed in summary table by default. (#1166)

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #1166

cc @aspina7

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed by running `renv::install()`
- [ ] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # gtsummary (development version)
 
-* Updated the default function in add_glance_*(galnce_fun=) for MICE models. (Update glance() method for mice objects in add_glance_table() #1137)
+* No longer removing the strata column from survey objects from the columns that will be summarized in `tbl_svysummary()`. If users previously didn't indicate which variables to summarize with `include=`, then the strata column (if present) will now appear in the summary table. When using a srvyr class object, all columns will be printed in summary table by default. (#1166)
```
Comment (larmarange) [inline NEWS.md:3]: Here the result will be different for a `survey` or a `srvyr` object. I would say that this is a complexity which is not relevant. The results should not change according to the type of the data set and both should be treated equally. I suspect that we will have misunderstanding if the rule differ for `survey` and `srvyr`.

I would be in favor in having a simple rule: all variables are, by default, returned by `everything()`. If user what a subset, they can use `include=` or `survey::subset()` or `srvyr::select()`.

In such case, we should simple remove `.remove_survey_cols()`

File: R/tbl_svysummary.R [hunk 2]
```diff
@@ -613,17 +613,11 @@ c_form <- function(left = NULL, right = 1) {
       select(-any_of(c(
         all.vars(x$call$id),
         all.vars(x$call$probs),
-        all.vars(x$call$strata),
```
Comment (larmarange) [inline R/tbl_svysummary.R:616]: cf. #1171 discussion. There are cases where we want to display `x$call$id`.
And in some (less frequent cases) we need to display statistics about `weights`. 

cf. previous comment. `everything()` should display all variables.

If really there is a need for identifying survey variables, let's create a `survey_vars()` selector and let the user indicates `include = -survey_vars()`

**Review by larmarange — CHANGES_REQUESTED** [review]
Hi @ddsjoberg 

Thanks a lot for the PR. I have the feeling that we should be simpler and treat equally `survey` and `srvyr` data sets.

**Comment by ddsjoberg** [comment]
thanks @larmarange , I updated to just print them all!

**Review by larmarange — APPROVED** [review]
Thanks a lot!! Thta's great

**Comment by aspina7** [comment]
Amazing thanks for moving so fast on this!

**Pull request merged**